### PR TITLE
Glharper/wav file size

### LINKED ISSUE #513: Missing chunkSize in speech audio file WAV format
We're using text-to-speech Microsoft Cognitive Services to generate speech audios. However we need it in WAV format according to following output format specifications:

```txt
Codec: PCMS16LE (araw)
Channel: Mono 
Sample Rate: 8000
Bits per Sample :16
```

Once we generate audio speech, we pass it to an IVR (Interactive Voice Response) service. When upload azure service generated speech audio file, we're getting error because of audio file format. We further dug into it, no clue until we inspect speech audio file metadata especially Raw Header. More information can be found here: https://github.com/Azure-Samples/cognitive-services-speech-sdk/issues/1450

After detailed analysis on different files Raw Headers, azure generated speech audio files with WAV format (Riff8Khz16BitMonoPcm) don't include chunkSize. We test it by converting speech audio through online tool provided by [3Cx](https://www.3cx.com/docs/converting-wav-file/) and the uploaded to IVR and it's working fine.

**Raw Header - Azure Speech API - success.wav**
```txt
52 49 46 46 00 00 00 00 57 41 56 45 66 6D 74 20 10 00 00 00 01 00 01 00 40 1F 00 00 80
3E 00 00 02 00 10 00 64 61 74 61 BE 1B 03 00 00 00 FF FF 00 00 01 00 FF FF 00 00 00 00
00 00 00 00 FF FF 00 00 FE FF FF FF 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
00 00 00 00 00 01 00 00 00 01 00 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
00 00 00 00 00 00 00 00 00 00 00 00
```
*[success.wav](https://storage.googleapis.com/pxd-audio-store/success.wav)*

**Raw Header - Converted converted_success.wav**
```txt
52 49 46 46 E2 1B 03 00 57 41 56 45 66 6D 74 20 10 00 00 00 01 00 01 00 40 1F 00 00 80 3E
00 00 02 00 10 00 64 61 74 61 BE 1B 03 00 00 00 FF FF 00 00 01 00 FF FF 00 00 00 00 00 00
00 00 FF FF 00 00 FE FF FF FF 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
00 00 01 00 00 00 01 00 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
00 00 00 00 00 00 00 00
```
*[converted_success.wav](https://storage.googleapis.com/pxd-audio-store/converted_success.wav)*

Tool that we used for conversion changes only chunkSize bytes from `00 00 00 00` to `E2 1B 03 00`

More Information about [WAVE PCM soundfile format](http://soundfile.sapp.org/doc/WaveFormat/)
image
*WAVE PCM soundfile format*

### PR DESCRIPTION
**Pull request by glharper**
Fix #513

### CODE REVIEW

**Review by yulin-li — APPROVED** [review]
:shipit:

File: src/sdk/Audio/AudioOutputFormat.ts [hunk 1]
```diff
@@ -532,6 +532,7 @@ export class AudioOutputFormatImpl extends AudioStreamFormatImpl {
     public updateHeader(audioLength: number): void {
         if (this.priHasHeader) {
             const view = new DataView(this.privHeader);
+            view.setUint32(4, audioLength + this.privHeader.byteLength, true);
```
Comment (yulin-li) [inline src/sdk/Audio/AudioOutputFormat.ts:535]: [This doc](http://soundfile.sapp.org/doc/WaveFormat/) says it should be `36 + audioLenght` while `this.privHeader.byteLength` is 44

Comment (yulin-li) [reply src/sdk/Audio/AudioOutputFormat.ts:535]: hmm, more accurate, it should be `this.privHeader.byteLength - 8 + audioLenght`

Comment (mudssrali) [reply src/sdk/Audio/AudioOutputFormat.ts:535]: Yes agree with you @yulin-li - it's mentioned that later one is more precise. I checked size variation by converting .wav to .wav with different engines. That's only possible how you're calculating size.

**Pull request merged**